sympy/tensor/indexed.py

```diff
@@ -156,7 +163,12 @@ def __new__(cls, base, *args, **kw_args):
             else:
                 return base[args]
 
-        return Expr.__new__(cls, base, *args, **kw_args)
+        obj = Expr.__new__(cls, base, *args, **kw_args)
```

Comment:
Were you going to wait on this feature (garbbing assumptions from the base) and do Function and IndexedBase at the same time?

Reply:
That was the plan, yes. I left `Indexed` inheriting from `IndexedBase` in there, thinking that still made sense to include here, but I realize that inadvertently also inherits from `Symbol` into `Indexed`...  I still think it makes sense to have the currently suggested feature, but the question is whether or not it should be introduced in this PR or the second one also fixing `Function` and `IndexedBase`. Or at this point, I could just add everything and update the PR description accordingly? Sorry about the mess.

Reply:
> a distinct behavior

totally agree.

Please double check my edits of the OP (and edit at will). Otherwise this looks ready to go.

Reply:
The edits look good to me, no objections 👍